Tried to build with docker-compose and push, but the push is running into issues. 

```
#34 DONE 0.1s
#35 [gateway] exporting to image
#35 exporting layers
#35 exporting layers 0.4s done
#35 writing image sha256:b080665af6f94968233aec8ec952b0c6c424576e2a271c42b5e9479e6fced0d7 done
#35 naming to docker.io/library/maeve-csms-gateway done
#35 DONE 0.4s

maeve-csms-gateway   latest      b080665af6f9   1 second ago     26.7MB
maeve-csms-manager   latest      f26bc6871258   12 seconds ago   39.5MB

maeve-csms-gateway                                latest      b080665af6f9   1 second ago     26.7MB
ghcr.io/everest/everest-demo/maeve-csms-gateway   0.0.16      b080665af6f9   1 second ago     26.7MB
maeve-csms-manager                                latest      f26bc6871258   12 seconds ago   39.5MB
ghcr.io/everest/everest-demo/maeve-csms-manager   0.0.16      f26bc6871258   12 seconds ago   39.5MB
moby/buildkit                                     latest      c6936aa67ee2   2 weeks ago      203MB

The push refers to repository [ghcr.io/everest/everest-demo/maeve-csms-gateway]

8fa10c0194df: Waiting
03af25190641: Waiting
denied: permission_denied: The requested installation does not exist.
Error: Process completed with exit code 1.
```

This seems to be because we don't have a package listed
<img width="1177" alt="Screenshot 2024-11-14 at 9 58 27 AM" src="https://github.com/user-attachments/assets/758fb8c6-cb6d-45c8-9ff5-9f9f37a11d08">

Switching to the build-and-push action at least initially since it looks like it creates it properly.
Will have to hardcode for now though.

--- a/.github/workflows/build_and_push_csms.yml
+++ b/.github/workflows/build_and_push_csms.yml
@@ -25,6 +25,15 @@ jobs:
           - csms: maeve
             csms_repo: louisg1337/maeve-csms
             csms_branch: set_charging_profile
+            image_name: manager
+            context: ./maeve-csms/manager
+            host_namespace: ghcr.io/everest/everest-demo
+          - csms: maeve
+            csms_repo: louisg1337/maeve-csms
+            csms_branch: set_charging_profile
+            image_name: gateway
+            context: ./maeve-csms/gateway
+            host_namespace: ghcr.io/everest/everest-demo
 #           - csms_dir: citrineos
 #             csms_repo: https://github.com/citrineos/citrineos-core.git
 #             csms_branch: feature/everest-demo
@@ -85,28 +94,42 @@ jobs:
             pushd ${{ matrix.csms }}-csms
             bash ../everest-demo/${{ matrix.csms }}/apply-patches.sh
 
-      - name: Build using docker compose
-        id: build-using-docker-compose
-        shell: bash
-        run: |
-            source everest-demo/.env
-            echo "Using docker-compose instead of the build-and-push action since we don't have to duplicate the set of services in that case"
-            pushd ${{ matrix.csms }}-csms
-            docker compose build
-            
-      - name: List, tag and push newly built images
-        id: list-tag-and-push-newly-built-images
-        shell: bash
-        run: |
-            source everest-demo/.env
-            echo "Before retagging, image list is"
-            docker image list
-            # hardcoding the list of containers for now
-            echo "Retagging {{ $matrix.csms }}-csms-gateway:latest to "
-            docker tag ${{ matrix.csms }}-csms-gateway:latest ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-gateway:"${TAG}"
-            docker tag ${{ matrix.csms }}-csms-manager:latest ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-manager:"${TAG}"
-            echo "After retagging, image list is"
-            docker image list
+#       - name: Build using docker compose
+#         id: build-using-docker-compose
+#         shell: bash
+#         run: |
+#             source everest-demo/.env
+#             echo "Using docker-compose instead of the build-and-push action since we don't have to duplicate the set of services in that case"
+#             pushd ${{ matrix.csms }}-csms
+#             docker compose build
+#             
+#       - name: List, tag and push newly built images
+#         id: list-tag-and-push-newly-built-images
+#         shell: bash
+#         run: |
+#             source everest-demo/.env
+#             echo "Before retagging, image list is"
+#             docker image list
+#             # hardcoding the list of containers for now
+#             echo "Retagging {{ $matrix.csms }}-csms-gateway:latest to "
+#             docker tag ${{ matrix.csms }}-csms-gateway:latest ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-gateway:"${TAG}"
+#             docker tag ${{ matrix.csms }}-csms-manager:latest ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-manager:"${TAG}"
+#             echo "After retagging, image list is"
+#             docker image list
+# 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set Docker image metadata
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ matrix.host_namespace }}/${{ matrix.image_name }}
+          tags: |
+            type=semver,pattern={{version}},value=v${{ steps.docker-image-version-check.outputs.TAG }}
 
       - name: Log into GitHub container registry
         uses: docker/login-action@v3
@@ -115,12 +138,22 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Push the images as packages
-        id: push-the-images-as-packages
-        shell: bash
-        run: |
-            source everest-demo/.env
-            docker push ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-gateway:"${TAG}"
-            docker push ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-manager:"${TAG}"
+#       - name: Push the images as packages
+#         id: push-the-images-as-packages
+#         shell: bash
+#         run: |
+#             source everest-demo/.env
+#             docker push ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-gateway:"${TAG}"
+#             docker push ghcr.io/everest/everest-demo/${{ matrix.csms }}-csms-manager:"${TAG}"
+# 
 
-
+      - name: Build and export to Docker
+        uses: docker/build-push-action@v6
+        with:
+          load: true
+          context: ${{ matrix.context }}
+          push: false
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          cache-from: type=gha,scope=${{ matrix.image_name }}
+          cache-to: type=gha,mode=max,scope=${{ matrix.image_name }}

--- a/.github/workflows/build_and_push_csms.yml
+++ b/.github/workflows/build_and_push_csms.yml
@@ -152,7 +152,7 @@ jobs:
         with:
           load: true
           context: ${{ matrix.context }}
-          push: false
+          push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           cache-from: type=gha,scope=${{ matrix.image_name }}

--- a/.github/workflows/build_and_push_csms.yml
+++ b/.github/workflows/build_and_push_csms.yml
@@ -18,6 +18,8 @@ jobs:
     permissions:
       contents: read
       packages: write
+      attestations: write
+      id-token: write
 
     strategy:
       matrix:

